sympy/integrals/manualintegrate.py

```diff
@@ -255,15 +272,14 @@ def possible_subterms(term):
             return r
         elif isinstance(term, Pow):
             r = []
-            if term.args[1].is_constant(symbol):
-                r.append(term.args[0])
-            elif term.args[0].is_constant(symbol):
-                r.append(term.args[1])
-            if term.args[1].is_Integer:
-                r.extend([term.args[0]**d for d in divisors(term.args[1])
+            for arg in term.args:
+                if not arg.is_constant(symbol):
+                    r.append(arg)
+            if term.exp.is_Integer:
+                r.extend([term.base**d for d in divisors(term.exp)
```

Comment:
I realise that you didn't write all of this code but there are a few things here that can be very slow and should generally be avoided.

The `is_constant` method is worse than calling `simplify` (in fact the first thing it does is call `simplify`). I'm pretty sure that whatever this is needed for here can be done with something much cheaper.

Also `divisors` can be extremely slow for large integers with many prime factors.

Reply:
I guess checking with `has` before doing anything else in `is_constant` should speed-up things at least a bit...

Reply:
Ahh, I read the code again and it was a bit naive of me to assume that it wasn't done.